x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/PushFiltersToSource.java

```diff
@@ -252,8 +251,6 @@ static boolean canPushToSource(Expression exp, LucenePushdownPredicates lucenePu
                 && Expressions.foldable(cidrMatch.matches());
         } else if (exp instanceof SpatialRelatesFunction spatial) {
             return canPushSpatialFunctionToSource(spatial, lucenePushdownPredicates);
-        } else if (exp instanceof Match mf) {
```

Comment:
These conditions are checked in validate() / type resolution in Match, so we can simplify

Reply:
I think that this is fine to remove. I'm also not sure that it was ever really necessary? Since it seems that the field and type were already checked in Match even before this change - which is fine, just trying to understand.

Reply:
It was unnecessary once we added the `Validatable` interface, where additional checks can be done once the fields have been resolved.